common/message_passing/client.h

```diff
@@ -23,9 +24,9 @@
 namespace message_passing {
 
 /**
- * @brief Handles sending messages to another node.
+ * @brief Handles connecting to another node and sending/receiving messages.
  */
-class Client {
+class Client : public Node {
  public:
   /**
    * @param thread_pool Thread pool to use internally for managing associated
```

Comment:
I'm not sure what you mean by "managing associated"

Reply:
Each `Client` instance requires a `SenderTask` and a `ReceiverTask`, which are the "associated tasks" to be managed.